tensorboard/backend/event_processing/db_import_multiplexer_test.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import os.path
+import shutil
+
+from tensorboard.backend.event_processing import db_import_multiplexer
+from tensorboard.backend import application
+import tensorflow as tf
+
+
+# A simple event with a summary that contains a single TensorProto of int32.
+RECORD = (b'\x0b\x00\x00\x00\x00\x00\x00\x00\x86\x15\xf5\x04*\t\n\x07B'
```

Comment:
These hardcoded 27 bytes are both really opaque and non-arbitrary. Can
you instead generate the appropriate data in a more readable way?

For instance, all your tests still pass if I replace the hard-coded file
writing with a normal `FileWriter` pipeline:

```diff
diff --git a/tensorboard/backend/event_processing/BUILD b/tensorboard/backend/event_processing/BUILD
index fddbe549..e7137ce0 100644
--- a/tensorboard/backend/event_processing/BUILD
+++ b/tensorboard/backend/event_processing/BUILD
@@ -204,6 +204,7 @@ py_test(
     srcs_version = "PY2AND3",
     deps = [
         ":db_import_multiplexer",
+        "//tensorboard/compat/proto:protos_all_py_pb2",
         "//tensorboard/backend:application",
         "//tensorboard:expect_tensorflow_installed",
     ],
diff --git a/tensorboard/backend/event_processing/db_import_multiplexer_test.py b/tensorboard/backend/event_processing/db_import_multiplexer_test.py
index 71d53a02..7ad2df48 100644
--- a/tensorboard/backend/event_processing/db_import_multiplexer_test.py
+++ b/tensorboard/backend/event_processing/db_import_multiplexer_test.py
@@ -23,20 +23,30 @@ import shutil
 
 from tensorboard.backend.event_processing import db_import_multiplexer
 from tensorboard.backend import application
+from tensorboard.compat.proto import event_pb2
+from tensorboard.compat.proto import summary_pb2
+from tensorboard.compat.proto import tensor_pb2
+from tensorboard.compat.proto import types_pb2
 import tensorflow as tf
 
 
-# A simple event with a summary that contains a single TensorProto of int32.
-RECORD = (b'\x0b\x00\x00\x00\x00\x00\x00\x00\x86\x15\xf5\x04*\t\n\x07B'
-          b'\x05\x08\x03:\x01\x01\x85\xb2\x08\x9c')
-
 def _AddEvents(path):
-  if not tf.gfile.IsDirectory(path):
-    tf.gfile.MakeDirs(path)
-  fpath = os.path.join(path, 'hypothetical.tfevents.out')
-  with tf.gfile.GFile(fpath, 'w') as f:
-    f.write(RECORD)
-    return fpath
+  with tf.summary.FileWriter(path) as writer:
+    event = event_pb2.Event(
+        summary=summary_pb2.Summary(
+          value=[
+            summary_pb2.Summary.Value(
+              tensor=tensor_pb2.TensorProto(
+                dtype=types_pb2.DT_INT32,
+                int_val=[1],
+              )
+            )
+          ]
+        )
+    )
+    native_event = tf.Event()
+    native_event.ParseFromString(event.SerializeToString())
+    writer.add_event(native_event)
 
 
 class DbImportMultiplexerTest(tf.test.TestCase):
```